src/traces/scattercarpet/plot.js

```diff
@@ -37,6 +38,6 @@ module.exports = function plot(gd, plotinfoproxy, data) {
         // separately to all scattercarpet traces, but that would require
         // lots of reorganization of scatter traces that is otherwise not
         // necessary. That makes this a potential optimization.
-        node.attr('clip-path', 'url(#clip' + carpet.uid + 'carpet)');
+        Drawing.setClipUrl(node, carpet._clipPathId);
```

Comment:
Ha. That means carpet traces will now be functional in Angular 1.0 see https://github.com/plotly/plotly.js/pull/509